jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java

```diff
@@ -128,11 +130,10 @@ private void addSecretsFromXmlConfiguration(KubernetesListBuilder builder) {
     }
 
     private List<SecretConfig> getSecretsFromXmlConfig() {
-        ResourceConfig resourceConfig = getConfiguration().getResource();
-        if(resourceConfig != null && resourceConfig.getSecrets() != null) {
-            return resourceConfig.getSecrets();
-        }
-        return null;
+        return Optional.ofNullable(getConfiguration().getResource())
+                .map(rc -> Optional.ofNullable(rc.getSecrets())
+                        .orElse(Collections.emptyList()))
+                .orElse(Collections.emptyList());
```

Comment:
@sankritimishra : Sorry for the previous suggestion, [Optional.map](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#map-java.util.function.Function-) takes care of null values by returning an empty optional. You can directly map result without using nested Optional.ofNullable
```suggestion
        return Optional.ofNullable(getConfiguration())
            .map(Configuration::getResource)
            .map(ResourceConfig::getSecrets)
            .orElse(Collections.emptyList());
```